cmd/process-agent/main_common.go

```diff
@@ -251,28 +224,36 @@ func runAgent(globalParams *command.GlobalParams, exit chan struct{}) {
 		_ = log.Error(err)
 	}
 
-	runApp(exit, syscfg, hostInfo)
+	runApp(exit, globalParams, hostInfo)
 
 	if err := srv.Shutdown(context.Background()); err != nil {
 		log.Errorf("Error shutting down expvar server on port %v: %v", expVarPort, err)
 	}
 }
 
-func runApp(exit chan struct{}, syscfg *sysconfig.Config, hostInfo *checks.HostInfo) {
+func runApp(exit chan struct{}, globalParams *command.GlobalParams, hostInfo *checks.HostInfo) {
 	go util.HandleSignals(exit)
 
-	var allChecks struct {
+	var appInitDeps struct {
 		fx.In
 		Checks []types.CheckComponent `group:"check"`
+		syscfg sysprobeconfig.Component
 	}
 	app := fx.New(
 		fx.Supply(
-			syscfg,
 			hostInfo,
+			core.BundleParams{
+				SysprobeConfigParams: sysprobeconfig.NewParams(
+					sysprobeconfig.WithSysProbeConfFilePath(globalParams.SysProbeConfFilePath),
+				),
+			},
 		),
-		fx.Populate(&allChecks),
+		// Populate dependencies required for initialization in this function.
+		fx.Populate(&appInitDeps),
```

Comment:
Yes. `appInitDeps` is used for two things:
- Preventing the agent from starting if there are no checks enabled. 
- Preventing the from starting if the user runs the agent with the (deprecated) `--info` command. This will likely go away in the future.

Since both of these rely on some components being initialized and determine whether the process agent runs or not (the ones in `appInitDeps`), we need to run them after the call to `fx.New` and before the call to `app.Start`.

I felt that these didn't really fit the singleton architecture of components very well, since they aren't wrapped in structs that other components may depend on, nor do they have a life cycle. Additionally, I am a bit worried that it may be hard to guarantee that this code runs before the call to `app.Start` if we were to put this code into components.

Implementation-wise, I also think it would be difficult to wrap this logic into components, since we bypass `fx`'s logic for blocking while the program is running, which means we can't use `Shutdowner` to stop the agent before it starts.